osu.Game/Overlays/Direct/DirectPanel.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenTK;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps.Drawables;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Overlays.Direct
+{
+    public abstract class DirectPanel : Container
+    {
+        protected readonly BeatmapSetInfo SetInfo;
+
+        protected DirectPanel(BeatmapSetInfo setInfo)
+        {
+            SetInfo = setInfo;
+        }
+
+        protected IEnumerable<DifficultyIcon> GetDifficultyIcons()
+        {
+            var icons = new List<DifficultyIcon>();
+
+            foreach (var b in SetInfo.Beatmaps)
+                icons.Add(new DifficultyIcon(b));
+
+            return icons;
+        }
+
+        protected Drawable GetBackground(TextureStore textures)
+        {
+            return new AsyncLoadWrapper(new Sprite
+            {
+                FillMode = FillMode.Fill,
+                Texture = new OnlineWorkingBeatmap(SetInfo.Beatmaps.FirstOrDefault(), textures, null).Background,

```

Comment:
This is evaluating `GetBackground` outside an async context. Check out `BeatmapBackgroundSprite` instead.